packages/babel-helper-create-class-features-plugin/src/fields.ts

```diff
@@ -510,16 +510,33 @@ function buildPrivateInstanceFieldInitSpec(
   ref: t.Expression,
   prop: NodePath<t.ClassPrivateProperty>,
   privateNamesMap: PrivateNamesMap,
+  state,
 ) {
   const { id } = privateNamesMap.get(prop.node.key.id.name);
   const value = prop.node.value || prop.scope.buildUndefinedNode();
 
-  return template.statement.ast`${t.cloneNode(id)}.set(${ref}, {
-    // configurable is always false for private elements
-    // enumerable is always false for private elements
-    writable: true,
-    value: ${value},
-  })`;
+  if (!process.env.BABEL_8_BREAKING) {
+    if (!state.availableHelper("classPrivateFieldInitSpec")) {
+      return template.statement.ast`${t.cloneNode(id)}.set(${ref}, {
+        // configurable is always false for private elements
+        // enumerable is always false for private elements
+        writable: true,
+        value: ${value},
+      })`;
+    }
+  }
+
+  const helper = state.addHelper("classPrivateFieldInitSpec");
+  const expr = t.callExpression(helper, [
+    t.thisExpression(),
+    t.cloneNode(id),
+    template.expression.ast`{
+      writable: true,
+      value: ${value}
+    }`,
+  ]);
+
+  return t.expressionStatement(expr);
```

Comment:
Nit: I think you can use template to generate the full call here:

```suggestion
  return template.statement.ast`${helper}(
    ${t.thisExpression()},
    ${id},
    {
      writable: true,
      value: ${value}
    },
  )`;
```

Reply:
The `id` above should still be cloned though.